### ISSUE TITLE
Ability to add a time-frame when applicable to query/response/client blocklists

### ISSUE THREAD

**Issue by cbuijs** [folbricht/routedns#158]
It would be nice to have the ability when a blocklist group would be in effect time wise. Default would what it is now (always active), the following should be great to have:

- Starting Time
- Ending Time
- Which Days of the week (Sun, Mon, Tue etc), Default is all days.

When Ending Time is before Starting Time, it is assumed the "next day" as seen from the starting-time.

This give kind of a nice "Parental Control" option to block certain things during the day when kids are "awake", or Breakfast, Lunch and Dinner-time to force screens off (yes, I am a parent :-) ).

This could be also part of a new `group`, where you can select which resolver next based on time, or part of `router` perhaps... Router makes more sense as you could centralize the "schedules" in one place kind of.

Example:
```
routes = [
    {starttime = "18:00:00", endtime = "19:00:00", days = ["Mon", "Tue", "Wed"], resolver = "family-safe"},
    {starttime = "22:00:00", endtime = "07:00:00", resolver = "nxdomain"},
    {resolver = "anything-goes"},
]
```
In above rudimentary/abstract example, at certain time a "family-safe" resolver is used to block adult-themes or game-sites/domains (for example). During the night everything is blocked, and if those conditions are not met, the default "anything-goes" (no filtering) resolver is used.

PS: This is just a "nice to have".

**Comment by folbricht** [comment]
Yes, good idea. I prefer the idea of a time-based router. That could be combined with any kind of other element, not just blocklist. Thinking if it's better in the existing router, or as a new element.

**Comment by cbuijs** [comment]
Ah, I just updated the original post with some examples. :-)
Yeah, that makes sense, doesn't really matter for me, I kind of liked the "routers" way. That is all.

**Comment by cbuijs** [comment]
BTW... There might be the need of a "flush-cache" feature on this. Because the TTL can go over the start-time for example and still serve answers maybe not wanted during the schedule.

One solution could be to use this always "before" cache (document).

Note: A flush-cache feature is cool anyway, also needed when lists are reloaded for example. I will make a separate post on/for that :-).

**Comment by folbricht** [comment]
Time-based routing is now available on the `issue-158` branch if you want to give it a try. It supports routing by day of the week, as well as time of day. Here a simple example:
```toml
[routers.router1]
routes = [
  { name = '(^|\.)twitter\.com\.$', weekdays = ["sat", "sun"], after = "09:00", before = "17:00", resolver="static-nxdomain" }, # No Twitter on weekends from 9am-5pm!
  { resolver="cloudflare-dot" }, # default route
]
```
This should provide a lot of flexibility as it can be combined with other elements. Not just blocklists or static responders.

**Comment by cbuijs** [comment]
Sweet! One question pops up immediately, I didn't test yet: What if before is before after? Is then "until next day" taken?

**Comment by folbricht** [comment]
Currently, something like `after = 21:00` and `before = 06:00` will cause the rule to never fire since the conditions can't be met. It only looks at the current day, so from `00:00` to `23:59` and doesn't consider the next or day before. Also, it's possible to only provide `before` or `after` which then just defaults the other param to the rest of the day.

I might be able to add a check that will error when a rule has the timestamps mixed up making it never fire.

**Comment by cbuijs** [comment]
No worries. It is just clarification and perhaps worthy a note in documentation.
